pkg/liveshare/rpc.go

```diff
@@ -40,7 +42,60 @@ func (r *rpcClient) do(ctx context.Context, method string, args, result interfac
 	return waiter.Wait(waitCtx, result)
 }
 
-type nullHandler struct{}
+type handlerFn func(conn *jsonrpc2.Conn, req *jsonrpc2.Request)
 
-func (nullHandler) Handle(ctx context.Context, conn *jsonrpc2.Conn, req *jsonrpc2.Request) {
+type requestHandler struct {
+	handlersMu sync.RWMutex
+	handlers   map[string][]handlerFn
+}
+
+func newRequestHandler() *requestHandler {
+	return &requestHandler{handlers: make(map[string][]handlerFn)}
+}
+
+func (r *requestHandler) register(requestType string, handler handlerFn) func() {
+	r.handlersMu.Lock()
+	defer r.handlersMu.Unlock()
+
+	if _, ok := r.handlers[requestType]; !ok {
+		r.handlers[requestType] = []handlerFn{}
```

Comment:
This isn't necessary. The zero value of r.handlers[requestType], nil, is a valid empty slice.

Reply:
Interesting. Learning a lot about Go with this PR 🙂 changed in latest commit